File: [프로그래머스] 코딩테스트 입문_캐릭터의 좌표.py
def solution(keyinput, board):
    answer = [0, 0]
    d = {"left":-1, "right":1, "up":1, "down":-1}
    for i in keyinput:
        if i in ["left", "right"]:
            answer[0] += d[i]
            if abs(answer[0]) > board[0]//2:
                answer[0] -= d[i]
        elif i in ["up", "down"]:
            answer[1] += d[i]
            if abs(answer[1]) > board[1]//2:
                answer[1] -= d[i]
    return answer

File: [프로그래머스] 코딩테스트 입문_캐릭터의 좌표.js
function solution(keyinput, board) {
    let answer = [0, 0];
    let d = {"left":-1, "right":1, "up":1, "down":-1}
    for (let i of keyinput){
         if (i == "left" || i == "right"){
             answer[0] += d[i];
             if (Math.abs(answer[0]) > parseInt(board[0]/2)){
                 answer[0] -= d[i];
            }
         } else if (i == "up" || i == "down"){
             answer[1] += d[i];
             if (Math.abs(answer[1]) > parseInt(board[1]/2)){
                 answer[1] -= d[i];
            }
         }
    }
    return answer;
}